geppetto-ui/src/components/big-image-viewer/BigImageViewer.js

```diff
@@ -0,0 +1,144 @@
+
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        this.loadViewer();
+    }
+
+    componentDidMount() {
+        this.loadViewer();
+    }
+
+    extractFilePath(data) {
+        let file;
+        if (data !== undefined) {
+            if (data.getMetaType === undefined) {
+                file = data;
+            } else if (data.getMetaType() === "Instance" && data.getVariable().getInitialValues()[0].value.format === "DZI") {
+                file = data.getVariable().getInitialValues()[0].value.data;
+            }
+        }
+        return file;
+    }
+
+    loadViewer() {
+        if (this.state.file !== undefined) {
+            if (this.viewer !== undefined) {
+                this.viewer.destroy();
+            }
+            this.state.settings.tileSources = this.state.file;
+            this.viewer = OpenSeaDragon(this.state.settings);
+        }
+    }
+
+    // These four methods are not exposed by OpenSeaDragon
+    goHome() {
+        this.viewer.viewport.goHome()
+    }
+
+    zoomIn() {
+        this.viewer.viewport.zoomBy(
+            this.viewer.zoomPerClick / 1.0
+        );
+        this.viewer.viewport.applyConstraints();
+
+    }
+
+    zoomOut() {
+        this.viewer.viewport.zoomBy(
+            1.0 / this.viewer.zoomPerClick
+        );
+        this.viewer.viewport.applyConstraints();
+    }
+
+    fullPage() {
+        this.viewer.setFullScreen(true);
+        this.viewer.fullPageButton.element.focus();
+        this.viewer.viewport.applyConstraints();
+    }
+
+    getCustomButtons() {
+        const customButtons = [];
+        customButtons.push({'icon': faSearchMinus, 'id': 'zoom-out', 'title': 'Zoom Out', 'action': this.zoomOut});
+        customButtons.push({'icon': faSearchPlus, 'id': 'zoom-in', 'title': 'Zoom In', 'action': this.zoomIn});
+        customButtons.push({'icon': faHome, 'id': 'home', 'title': 'Center Image', 'action': this.goHome});
+        return customButtons;
+    }
+
+    render() {
+        const {classes} = this.props;
+        const customButtons = this.getCustomButtons();
+
+        return (
+            <div className={classes.bigImageViewer}>
+                <Toolbar className={classes.toolbar}>
+                    <div className={classes.toolbarBox}>
+                        {customButtons.map(customButton =>
+                            <IconButtonWithTooltip
+                                key={customButton.id}
+                                disabled={false}
+                                onClick={() => customButton.action()}
+                                className={classes.button}
+                                tooltip={customButton.title}
+                                icon={customButton.icon}
+                            />
+                        )}
+                    </div>
+                </Toolbar>
+                <div id={this.props.id + "_component"} className={classes.bigImageViewer}/>
```

Comment:
Better avoid ids, too easy to duplicate

Reply:
I fear it is necessary due to the use of [OpenSeaDragon](https://github.com/openseadragon/openseadragon) (which requires the id of the element where the zoomable image will be added) by the BigImageViewer

Reply:
Another case where I'm using ids is to add a key to each element of react arrays:
![image](https://user-images.githubusercontent.com/19196034/78822213-99e16500-79d2-11ea-91be-4d1a05a9c8b8.png)